### PR TITLE
SW-1559 Update storage location handling in v2 API

### PR DESCRIPTION
**Pull request by sgrimm**
The web app needs to be able to set the accession's storage location at creation
time, and no longer has a concept of "storage conditions." Update the API
accordingly.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #586** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/586" target="_blank">Graphite</a> 
    * **PR #587** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/587" target="_blank">Graphite</a> 
      * **PR #588** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/588" target="_blank">Graphite</a> 
        * **PR #589** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/589" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/589?utm_source=stack-comment).

**Review by karthikbtf — APPROVED** [review]
Thank you so much!

**Comment by karthikbtf** [comment]
@sgrimm this one will unblock a few FE PRs that are currently waiting for this to land.

**Comment by sgrimm** [comment]
**Graphite Merge Job**

Current status: ✅ Merged

This pull request was successfully merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/589?utm_source=stack-comment).


Job Reference: QpIP1QCR1Nvluk7Y6jyS

**Pull request merged**
